### PR TITLE
feat(client): Various improvements to config operations

### PR DESCRIPTION
**Pull request by nathansamson**
This includes 2 improvements to config operations.

Replaces #3798 

Note that this version is not manually tested by me, as it is not compatible with my cluster. As the changes are in code that seemingly didn't change since 1.6.1 I don't expect any troubles, but you never know...

### CODE REVIEW

File: client/deis.py [hunk 1]
```diff
@@ -1348,6 +1335,24 @@ def _read_config_from_path(self, path):
 
         return env_dict
 
+    def _display_configuration(self, configuration, oneline=False, highlight=[]):
```
Comment (bacongobbler) [inline client/deis.py]: let's make the `oneline` and `highlight` options as [keyword arguments](http://stackoverflow.com/a/1419160) to make this a little easier to maintain.

**Comment by carmstrong** [comment]
@nathansamson Are you able to make the change @bacongobbler requested? Thanks for taking the time to submit this!

**Comment by nathansamson** [comment]
@carmstrong done. Somehow I had missed the comment

File: client/deis.py [hunk 2]
```diff
@@ -1328,8 +1316,7 @@ def config_unset(self, args):
             if len(items) == 0:
                 self._logger.info('No configuration')
                 return
-            for k, v in values.items():
-                self._logger.info("{k}: {v}".format(**locals()))
+            self._display_configuration(values, oneline=False)
```
Comment (bacongobbler) [inline client/deis.py]: `oneline` is set to False by default so we can remove the invocation from here.

File: client/deis.py [hunk 3]
```diff
@@ -1348,6 +1335,27 @@ def _read_config_from_path(self, path):
...
+        highlight = kwargs.get('highlight', [])
+
+        keys = sorted(configuration)
+        if not oneline:
+            width = max(map(len, keys)) + 5
+            for k in keys:
+                k, v = encode(k), encode(configuration[k])
+                key = "{k:<" + str(width) + "}"
+                if k in highlight:
+                    self._logger.info(("\033[1m" + key + "\033[0m {v}").format(**locals()))
```
Comment (bacongobbler) [inline client/deis.py]: let's use the `colored` library as we do here for better portability: https://github.com/deis/deis/blob/master/client/deis.py#L682-L691

**Comment by mboersma** [comment]
Highlighting the config that actually changed is nice. (I added asterisks to indicate where bold green ANSI highlighting was applied.)
```
$ deis config:set FOO=bar
Creating config... done, v2

=== outlaw-aqualung
*FOO*    bar
$ deis config:set BOO=far
Creating config... done, v3

=== outlaw-aqualung
*BOO*    far
FOO      bar
$ deis config:set FOO=baz
Creating config... done, v4

=== outlaw-aqualung
BOO      far
*FOO*    baz
$ deis config:list
=== outlaw-aqualung Config
BOO      far
FOO      baz
$ deis config:set FOO=bax BAX=foo
Creating config... done, v5

=== outlaw-aqualung
*BAX*    foo
BOO      far
*FOO*    bax
```

If this PR could use the termcolor library as suggested I think it's ready to go. Something like:
```python
self._logger.info("{} {}".format(colored(key, 'green'), v))
```

**Comment by bacongobbler** [comment]
s/green/magenta. Deis needs more pink! (kidding)

**Comment by mboersma** [comment]
The second commit should use a lower-case verb as per our [code standards](http://docs.deis.io/en/latest/contributing/standards/#subject-line):
```
feat(client): highlight changed config
```

**Comment by nathansamson** [comment]
Hi,

I will close this pull request in the hope that @Joshua-Anderson will include something similar in #4015 

Otherwise I might try a third attempt in the new codebase

Regards,
Nathan

**Pull request closed without merging**